plugins/by-name/crates/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, ... }:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "crates";
+  packPathName = "crates.nvim";
+  package = "aerial-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  imports =
+    [ { cmpSourcePlugins.crates = "crates"; } ]
+    ++
+    # TODO introduced 2024-12-12: remove after 25.05
+    lib.nixvim.mkSettingsRenamedOptionModules [ "plugins" "crates-nvim" ]
+      [ "plugins" "crates" ]
+      [
+        "enable"
+        "package"
+        {
+          old = "extraOptions";
+          new = "settings";
+        }
+      ];
```

Comment:
This can be simplified to use `optionsRenamedToSettings`, right?

Reply:
Oh, nevermind. I missed that it's doing the plugin rename

Reply:
I'm not sure ottomh if this works. Doesn't `mkSettingsRenamedOptionModules` automatically nest into `settings`?

Reply:
No, I need to also rename `enable` and `package` . This is one level upper.

Reply:
> I'm not sure ottomh if this works. Doesn't `mkSettingsRenamedOptionModules` automatically nest into `settings`?

No. It takes two paths `old` and `new. Then, it does all the renames using the provided list of sub-paths, each being relative to `old` (for the rename source) and `new` (for the rename destination).